storage/src/cache/cachedfile.rs

```diff
@@ -137,17 +196,26 @@ impl BlobCache for FileCacheEntry {
         bios: &[BlobIoDesc],
     ) -> StorageResult<usize> {
         let mut bios = bios.to_vec();
+        let mut fail = false;
         bios.iter_mut().for_each(|b| {
-            if let Some(chunks_meta) = &self.meta {
-                // TODO: the first blob backend io triggers chunks array download.
-                if let BlobIoChunk::Address(_blob_index, chunk_index) = b.chunkinfo {
-                    let cki = BlobMetaChunk::new(chunk_index as usize, &chunks_meta.state);
-                    b.chunkinfo = BlobIoChunk::Base(Arc::new(cki));
+            if let BlobIoChunk::Address(_blob_index, chunk_index) = b.chunkinfo {
+                if let Some(meta) = self.meta.as_ref() {
+                    if let Some(bm) = meta.get_blob_meta() {
+                        let cki = BlobMetaChunk::new(chunk_index as usize, &bm.state);
+                        b.chunkinfo = BlobIoChunk::Base(Arc::new(cki));
+                    } else {
+                        warn!("failed to get blob.meta for prefetch");
+                        fail = true;
```

Comment:
Can we just return error from here>

Reply:
We are within an "for_each()" loop and can't return error here.